From 7857e5aa2459b2746e062ae59ae0240c4af7af5d Mon Sep 17 00:00:00 2001
From: Tom Hochstein <tom.hochstein@nxp.com>
Date: Mon, 29 Oct 2018 21:40:32 +0000
Subject: [PATCH] g2d-renderer: Fix open function build break

|                  from ../git/libweston/g2d-renderer.c:40:
| In function 'open',
|     inlined from 'g2d_renderer_create' at ../git/libweston/g2d-renderer.c:1629:2:
| /home/r60874/upstream/fsl-xwayland/tmp/work/cortexa9t2hf-neon-mx6qdl-fsl-linux-gnueabi/weston/4.0.0.imx-r0/recipe-                                                                        sysroot/usr/include/bits/fcntl2.h:50:4: error: call to '__open_missing_mode' declared with attribute error: open wit                                                                        h O_CREAT or O_TMPFILE in second argument needs 3 arguments
|     __open_missing_mode ();
|     ^~~~~~~~~~~~~~~~~~~~~~
| Makefile:5266: recipe for target 'libweston/g2d_renderer_la-g2d-renderer.lo' failed

Upstream-Status: Inappropriate [i.MX-specific]

Signed-off-by: Tom Hochstein <tom.hochstein@nxp.com>
---
 libweston/g2d-renderer.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

Index: git/libweston/g2d-renderer.c
===================================================================
--- git.orig/libweston/g2d-renderer.c
+++ git/libweston/g2d-renderer.c
@@ -1626,7 +1626,7 @@ g2d_renderer_create(struct weston_compos
 	path = malloc(strlen(dir) + 40);
 	strcpy(path, dir);
 	strcat(path, "/use-g2d-renderer");
-	close(open(path, O_CREAT | O_RDWR));
+	close(open(path, O_CREAT | O_RDWR, 0600));
 	free(path);
 
 	return 0;
